source/DAGView.cs

```diff
@@ -507,10 +507,6 @@ private void pathExplanationButton_Click(object sender, EventArgs e)
                         highlightPath(cyclePath);
                         _z3AxiomProfiler.UpdateSync(cyclePath);
                         toRemove = cyclePath.GetInstnationsUnusedInGeneralization();
-                        if (cyclePath.NeedsIds())

```

Comment:
Out of interest, when does this function return true?

Reply:
NeedsIds() is true when CycleDetection.NeedsIds is true, explained in the other conversation